test/api-binder.ts

```diff
@@ -0,0 +1,44 @@
+import { expect } from 'chai';
+import { APIBinder, APIBinderConstructOpts } from '../src/api-binder';
+import { DeviceApplicationState } from '../src/types/state';
+
+describe.only('APIBinder', () => {
+	let apiBinder: APIBinder;
+
+	before(() => {
+		apiBinder = new APIBinder({} as APIBinderConstructOpts);
```

Comment:
Does this cast actually work? I would expect it to fail if this is actually being type checked

Reply:
It does, the test passes.

Reply:
@CameronDiver sorry, I'm not deeply experienced with TypeScript details :), would you elaborate how this can be written in a different way?

Also, I have to disagree with your :)
>This shouldn't work

I does work: the code does get compiled, as you can see from build results.

If understand this correctly, TypeScript allows casts to interfaces as long as the dictionary is a subset of interface properties. 

For instance, from a [different commit](https://github.com/balena-io/balena-supervisor/commit/75863ccf79cf45fad95157eef18b037b12c4cdbf):
```
{Id: 'image-1'} as Docker.ImageInfo
```
This dictionary keys form a subset of `ImageInfo` interface properties, and the cast works.

In this case, we have an empty set, which is a subset of any set ;)

Reply:
Ugh, this wan't something I was aware of: https://github.com/Microsoft/TypeScript/issues/13788

Pretty awful. Anyway fair enough, it's just tests.